app/ts/background/background.ts

```diff
@@ -62,13 +62,22 @@ export const simulateGovernanceContractExecution = async (pendingTransaction: Pe
 		const pendingResults = pendingTransaction.simulationResults
 		if (pendingResults.statusCode !== 'success') return returnError('Voting transaction failed')
 		const fourByte = get4Byte(pendingTransaction.transactionToSimulate.transaction.input)
-		if (fourByte === undefined) return returnError('Could not identify the 4byte signature')
+		const fourByteString = get4ByteString(pendingTransaction.transactionToSimulate.transaction.input)
+		if (fourByte === undefined || fourByteString === undefined) return returnError('Could not identify the 4byte signature')
 		const explanation = FourByteExplanations[fourByte]
-		if ((explanation !== 'Submit vote' && explanation !== 'Cast vote') || pendingResults.data.simulatedAndVisualizedTransactions[0]?.events.length !== 1) return returnError('Could not identify the transaction as a vote')
+		if ((explanation !== 'Cast Vote'
+			&& explanation !== 'Submit Vote'
+			&& explanation !== 'Cast Vote By Sig'
+			&& explanation !== 'Cast Vote With Reason'
+			&& explanation !== 'Cast Vote With Reason And Params'
+			&& explanation !== 'Cast Vote With Reason And Params By Sig')
+			|| pendingResults.data.simulatedAndVisualizedTransactions[0]?.events.length !== 1) return returnError('Could not identify the transaction as a vote')
```

Comment:
If you accept suggestion for rewording some of these and don't get an error here, then we should fix it so this errors if you try to match against impossible strings.